src/Orc.WorkspaceManagement/Managers/WorkspaceManager.cs

```diff
@@ -117,40 +119,18 @@ public IWorkspace Workspace
 
             Initializing.SafeInvoke(this);
 
-            if (Directory.Exists(baseDirectory))
-            {
-                foreach (var workspaceFile in Directory.GetFiles(baseDirectory, string.Format("*{0}", WorkspaceFileExtension)))
-                {
-                    try
-                    {
-                        Log.Debug("Loading workspace from '{0}'", workspaceFile);
-
-                        using (var fileStream = new FileStream(workspaceFile, FileMode.Open))
-                        {
-                            var workspace = ModelBase.Load<Workspace>(fileStream, SerializationMode.Xml);
-                            if (workspace == null || string.IsNullOrEmpty(workspace.Title))
-                            {
-                                Log.Warning("File '{0}' doesn't look like a workspace, ignoring file", workspaceFile);
-                            }
-                            else
-                            {
-                                _workspaces.Add(workspace);
-
-                                Log.Debug("Loaded workspace");
-                            }
-                        }
-                    }
-                    catch (Exception ex)
-                    {
-                        Log.Error(ex, "Failed to load workspace from '{0}'", workspaceFile);
-                    }
-                }
-            }
+            _workspaces.Clear();
+
+            _workspaces.AddRange(_workspacesStorageService.LoadWorkspaces(baseDirectory));            
```

Comment:
Try not to put too many calls on the same line. Split them up (easier to debug and understand for next developers)